utils/ai.py

```diff
@@ -208,32 +188,73 @@ def get_local_chat_response(messages):
-        data = {
-            "messages": messages,
-            "temperature": 0.2,
-            "max_tokens": -1,
-            "stream": True,
-        }
-        response = requests.post(
-            url, headers=headers, data=json.dumps(data), stream=True, timeout=120
-        )
+        if model == "local-model":
+            url = "http://localhost:1234/v1/chat/completions"
+            headers = {"Content-Type": "application/json"}
+            data = {
+                "messages": messages,
+                "temperature": 0.2,
+                "max_tokens": -1,
+                "stream": True,
+            }
+            response = requests.post(
+                url, headers=headers, data=json.dumps(data), stream=True, timeout=120
+            )
 
-        if response.status_code == 200:
-            for chunk in response.iter_content(chunk_size=None):
-                decoded_chunk = chunk.decode()
-                if (
-                    "data:" in decoded_chunk and decoded_chunk.split("data:")[1].strip()
-                ):  # Check if the chunk is not empty
-                    try:
-                        chunk_dict = json.loads(decoded_chunk.split("data:")[1].strip())
-                        yield chunk_dict["choices"][0]["delta"].get("content", "")
-                    except json.JSONDecodeError:
-                        pass
+            if response.status_code == 200:
+                for chunk in response.iter_content(chunk_size=None):
+                    decoded_chunk = chunk.decode()
+                    if (
+                        "data:" in decoded_chunk
+                        and decoded_chunk.split("data:")[1].strip()
+                    ):  # Check if the chunk is not empty
+                        try:
+                            chunk_dict = json.loads(
+                                decoded_chunk.split("data:")[1].strip()
+                            )
+                            yield chunk_dict["choices"][0]["delta"].get("content", "")
+                        except json.JSONDecodeError:
+                            pass
+            else:
+                print(f"Error: {response.status_code}, {response.text}")
+                raise Exception("Internal Server Error")
         else:
-            print(f"Error: {response.status_code}, {response.text}")
-            raise Exception("Internal Server Error")
+            if not os.environ.get("OPENROUTER_API_KEY"):
+                raise Exception(
+                    f"For non-OpenAI models, like {model}, set your OPENROUTER_API_KEY."
+                )
+
+            response = requests.post(
+                url="https://openrouter.ai/api/v1/chat/completions",
+                headers={
+                    "Authorization": f"Bearer {os.environ.get('OPENROUTER_API_KEY')}",
+                    "HTTP-Referer": os.environ.get(
+                        "OPENROUTER_APP_URL", "https://fleet.so/context"
```

Comment:
To be consistent with the app title setting below, this should read from OPENROUTER_APP_URL if set in the env otherwise default to fleet